java/kotlin-extractor/src/main/kotlin/KotlinFileExtractor.kt

```diff
@@ -823,10 +824,60 @@ open class KotlinFileExtractor(
         }
     }
 
+    private fun isKotlinDefinedInterface(cls: IrClass?) =
+        cls != null && cls.isInterface && cls.origin != IrDeclarationOrigin.IR_EXTERNAL_JAVA_DECLARATION_STUB
```

Comment:
Do we need to cover annotation classes?

Reply:
No, because they cannot have non-Object supertypes or members, so the circumstance should never be able to arise.